cmd/ethkey/main.go

```diff
@@ -0,0 +1,322 @@
+package main
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/pborman/uuid"
+	"gopkg.in/urfave/cli.v1"
+)
+
+const (
+	version   = "0.1.0"
+	copyright = "Copyright 2017 The go-ethereum Authors"
+
+	defaultKeyfilePath = "keyfile.json"
+)
+
+var (
+	app *cli.App // the main app instance
+)
+
+var ( // Commonly used command line flags.
+	passphraseFlag = cli.StringFlag{
+		Name:  "passphrase",
+		Usage: "The passphrase for the keyfile.",
+	}
+
+	passphrasefileFlag = cli.StringFlag{
+		Name:  "passwordfile",
+		Usage: "The file that contains the passphrase for the keyfile.",
+	}
+)
+
+// Configure the app instance.
+func init() {
+	app = cli.NewApp()
+	app.Name = filepath.Base(os.Args[0])
+	app.Version = version
+	app.Copyright = copyright
+	app.Usage = "an Ethereum key manager"
+	app.Commands = []cli.Command{
+		{
+			Action:    generate,
+			Name:      "generate",
+			Usage:     "generate new keyfile",
+			ArgsUsage: "<keyfile>",
+			Description: `
+Generate a new keyfile.
+`,
+			Flags: []cli.Flag{
+				passphraseFlag,
+				passphrasefileFlag,
+			},
+		},
+		{
+			Action:    inspect,
+			Name:      "inspect",
+			Usage:     "inspect a keyfile",
+			ArgsUsage: "<keyfile>",
+			Description: `
+Print various information about the keyfile.
+`,
+			Flags: []cli.Flag{
+				passphraseFlag,
+				passphrasefileFlag,
+			},
+		},
+		{
+			Action:    sign,
+			Name:      "sign",
+			Usage:     "sign a message",
+			ArgsUsage: "<keyfile-privkey> <message>",
+			Description: `
+Sign the message with a private key.
+Either a keyfile or a private key in hexadecimal encoding can be provided.
+If the message is an existing file, the content of the file will be used as the message.
+`,
+			Flags: []cli.Flag{
+				passphraseFlag,
+				passphrasefileFlag,
+			},
+		},
+		{
+			Action:    verify,
+			Name:      "verify",
+			Usage:     "verify the signature of a message",
+			ArgsUsage: "<keyfile-address-pubkey> <signature> <message>",
+			Description: `
+Verify the signature of the message. 
+Either a keyfile, address or public key can be provided.
+If the message is an existing file, the content of the file will be used as the message.
+`,
+			Flags: []cli.Flag{},
+		},
+	}
+}
+
+// generate implements the generate command.
+func generate(ctx *cli.Context) error {
+	passphrase := getPassPhrase(ctx, true)
+
+	// Check if keyfile path given and make sure it doesn't already exist.
+	keyfilepath := ctx.Args().First()
+	if keyfilepath == "" {
+		keyfilepath = defaultKeyfilePath
+	}
+	var err error
+	keyfilepath, err = filepath.Abs(keyfilepath)
```

Comment:
Why do you need the absolute path?

Reply:
`os.Stat` was failing for some reason while I was testing. I tried without and it works fine, so I remove all the `filepath.Abs` calls. Thanks for the critical suggestion :)